### PR TITLE
Routing topic

### LINKED ISSUE #72: Add Routing topic
We should move the routing content out of the components doc into a separate routing topic.

- Router component
- `@page`
- `NavLink` component

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #72

### CODE REVIEW

**Comment by danroth27** [comment]
@guardrex This PR has been sitting around for a while. Does it just need to be reviewed? Looks like it needs to be rebased.

**Comment by guardrex** [comment]
@danroth27 It just needs a review.

**Comment by guardrex** [comment]
@SteveSandersonMS, @rynowak, @javiercn Is anyone free for a few minutes to take a look at the open PRs?

* This PR and https://github.com/aspnet/Blazor.Docs/pull/101/files have been here for six weeks.
* I'd really like to get the language on component attributes and the `bind` Tag Helper fixed up on https://github.com/aspnet/Blazor.Docs/pull/147/files.
* The JS interop PR is fairly short, so it shouldn't take long. https://github.com/aspnet/Blazor.Docs/pull/151/files

File: docs/routing.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+[!INCLUDE[](~/includes/blazor-preview-notice.md)]
+
+Learn how to route requests in a client-side Blazor app and about the NavLink component.
+
+[View or download sample code](https://github.com/aspnet/Blazor.Docs/tree/master/docs/components/common/samples/) ([how to download](xref:client-side/blazor/index#view-and-download-samples)). See the [Get started](xref:client-side/blazor/get-started) topic for prerequisites.
+
+## Route templates
+
+Routing in Blazor is achieved by providing a route template to each accessible component in the app.
```
Comment (danroth27) [inline docs/routing.md]: Routing only works if you first add the Router component to your app. In our templates this is done in *App.cshtml*

```html
<Router AppAssembly=typeof(Program).Assembly />
```

Setting up the Router component will probably move to *Program.cs* some day, but for now this is how it's done.

File: docs/routing.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Multiple route templates can be applied to a component. In the [sample app](https://github.com/aspnet/Blazor.Docs/tree/master/docs/components/common/samples/), the following component responds to requests for `/BlazorRoute` and `/DifferentBlazorRoute`.
+
+*Pages/BlazorRoute.cshtml*:
+
+[!code-cshtml[](components/common/samples/2.x/ComponentsSample/Pages/BlazorRoute.cshtml?start=1&end=4)]
+
+## Route parameters
+
+Blazor components can receive route parameters from the route template provided in the `@page` directive. The Blazor client-side router uses route parameters to populate the corresponding component parameters.
```
Comment (danroth27) [inline docs/routing.md]: The Blazor client-side router uses route parameters to populate the corresponding component parameters with the same name (case insensitive).

File: docs/routing.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+*Pages/RouteParameter.cshtml*:
+
+[!code-cshtml[](components/common/samples/2.x/ComponentsSample/Pages/RouteParameter.cshtml?start=1&end=8)]
+
+Optional parameters aren't supported, so two `@page` directives are applied in the example above. The first permits navigation to the component without a parameter. The second `@page` directive takes the `{text}` route parameter and assigns the value to the `Text` property.
```
Comment (danroth27) [inline docs/routing.md]: Optional parameters aren't supported yet

Comment (guardrex) [reply docs/routing.md]: I say that, and I'm explaining why there are two `@page` directives present and how the workaround works. Are you saying u want this removed?

**Comment by guardrex** [comment]
@danroth27 Updates complete :guitar:

Check Line 26 tho on the Router component to see if that's what you want there.

**Pull request merged**